lib/ansible/modules/cloud/amazon/emr_cluster.py

```diff
@@ -0,0 +1,363 @@
+  - aws
+  - ec2
+'''
+
+EXAMPLES = '''
+- emr_cluster:
+    name: emr-jobs
+    region: us-east-1
+    log_uri: s3://bucket-name/logs/emr
+    wait: yes
+    applications:
+      - name: Hadoop
+      - name: Spark
+    steps:
+      - name: s3-dist-cp
+        hadoop_jar_step:
+          jar: command-runner.jar
+          action_on_failure: TERMINATE_CLUSTER
+          args:
+            - "s3-dist-cp"
+            - "--src=s3://bucket-name/events"
+            - "--dest=hdfs://aggregate"
+            - "--groupBy=.*/[0-9]{4}/[0-9]{2}/.*gz"
+            - "--targetSize=128"
+            - "--outputCode=snappy"
+    instances:
+      ec2_key_name: ec2_key
+      ec2_subnet_id: subnet-923a31db
+      configurations:
+        - classification: spark
+          properties:
+            maximizeResourceAllication: true
+      additional_slave_security_groups:
+        - sg-8eb2b1f1
+      additional_master_security_groups:
+        - sg-8eb2b1f1
+      instance_groups:
+        - name: master
+          instance_role: MASTER
+          instance_count: 1
+          instance_type: m3.xlarge
+          market: ON_DEMAND
+        - name: core
+          instance_role: CORE
+          instance_count: 4
+          instance_type: m3.xlarge
+          market: ON_DEMAND
+'''
+
+RETURN = '''
+# For more information see http://boto3.readthedocs.io/en/latest/reference/services/emr.html#EMR.Client.describe_cluster
+---
+cluster:
+  description: dictionary containing informations about EMR Cluster
+  returned: success
+  type: complex
+  contains:
+    id:
+      description: the unique id for the cluster
+      type: str
+      sample: j-2JNVMF38MZVHT
+    name:
+      description: the name of the cluster
+      type: str
+      sample: emr-jobs
+    status:
+      description: the current status defailts about the cluster
+      type: dict
+    ec2_instaces_attributes:
+      description: provides information about ec2
+      type: dict
+    log_uri:
+      description: path for logs location in S3
+      type: str
+    release_label:
+      description: the release label for the Amazon EMR
+      type: str
+    auto_terminate:
+      description: if true cluster should terminate after completing all step
+      type: boolean
+    applications:
+      description: list of the applications installed on this cluster
+      type: list
+    tags:
+      description: list of tags
+      type: dict
+    service_role:
+      description: IAM role that will be assumed by the Amazon AEMR service
+      type: str
+    master_public_dns_name:
+      description: the dns name of the master node
+      type: str
+    configurations:
+      description: the list of configurations supplied to EMR cluster
+      type: dict
+'''
+
+try:
+    from botocore.exception import ClientError
+except ImportError:
+    pass  # will be picked up from imported HAS_BOTO3
+
+import re
+import traceback
+import time
+
+from ansible.module_utils.basic import AnsibleModule
```

Comment:
This can be updated to AnsibleAWSModule from ansible.module_utils.aws.core

That will also mean you can remove the imports and uses of get_aws_connection_info, boto3_conn, and HAS_BOTO3. You can create the client with `connection = module.client('emr')` (which does not need exception handling around it)